### ISSUE TITLE
Averaged SGD

### ISSUE THREAD

**Issue by MaxHalford** [creme-ml/creme#182]
Leon Bottou discusses it [here](https://leon.bottou.org/publications/pdf/compstat-2010.pdf) while PyTorch has implemented it [here](https://pytorch.org/docs/stable/_modules/torch/optim/asgd.html). Ideally I think we should implement an `Averager` class which takes as input any optimizer. This way we can do averaging on SGD by doing `Averager(SGD(0.01))` but also on Adam, AdaGrad, etc.

I tried to implement averaged stochastic gradient descent but I ran into some API issues. I'm too tired to solve it straightaway so I rather open an issue. Feel free to take a look at it.
